### DISCUSSION TITLE
On the issue of energy not converging

### DISCUSSION THREAD

**Discussion by nzynpu** [deepmodeling/dpgen#920] [Q&A] [answered]
After I finished training using dpgen, by reading the energy .raw in 02.fp and comparing it with the energy.npy file, it was found that the energy calculated by dpgen was very different from the energy calculated by DFT, and there was no convergence.What is the reason for this？thanks.

The param.jason file is as follows

{
"type_map": ["Ba","Ti","S"],
"mass_map": [137.33, 47.87, 32.06],
"_comment": "initial data set for Training and the number of frames in each training batch",
"init_data_prefix": "./",
"init_data_sys": ["P1-POSCAR/02.md/sys-0002-0002-0006/deepmd"]

"sys_configs_prefix": "P1-POSCAR/02.md/sys-0002-0002-0006",
"sys_configs": [
["scale-1.000/00000*/POSCAR"],
["scale-1.000/00001*/POSCAR"],
["scale-1.000/00002*/POSCAR"],
["scale-1.000/00003*/POSCAR"]
],
"sys_batch_size": [8, 8, 8, 8],
"_comment": " 00.train ",
"numb_models": 4,
"default_training_param": {
"model": {
"type_map": ["Ba","Ti","S"],
"descriptor": {
"type": "se_a",
"sel": [36,36,108],
"rcut_smth": 0.5,
"rcut": 5.0,
"neuron": [160,160,160],
"resnet_dt": true,
"axis_neuron": 12,
"seed": 1
},
"fitting_net": {
"neuron": [120,120,120],
"resnet_dt": true,
"coord_norm": true,
"type_fitting_net": false,
"seed": 1
}
},
"loss": {
"start_pref_e": 0.1,
"limit_pref_e": 0.1,
"start_pref_f": 1000,
"limit_pref_f": 1,
"start_pref_v": 0,
"limit_pref_v": 0
},
"learning_rate": {
"type": "exp",
"start_lr": 0.005,
"decay_steps": 1000,
"decay_rate": 0.95
},
"training": {
"systems": [],
"set_prefix": "set",
"stop_batch":100000,
"batch_size": 1,
"seed": 1,
"_comment": "frequencies counted in batch",
"disp_file": "lcurve.out",
"disp_freq": 1000,
"numb_test": 100,
"save_freq": 1000,
"save_ckpt": "model.ckpt",
"load_ckpt": "model.ckpt",
"disp_training": true,
"time_training": true,
"profiling": false,
"profiling_file": "timeline.json"
}
},
"_comment": " 01.model_devi ",
"model_devi_dt": 0.002,
"model_devi_skip": 0,
"model_devi_f_trust_lo": 0.05,
"model_devi_f_trust_hi": 0.15,
"model_devi_clean_traj": false,
"model_devi_jobs": [
{"sys_idx": [0],"temps": [300],"press": [1],"trj_freq": 10,"nsteps": 3000, "ensemble": "nvt", "_idx": "00"
},
{"sys_idx": [1],"temps": [300],"press": [1],"trj_freq": 10,"nsteps": 6000, "ensemble": "nvt", "_idx": "01"
},
{"sys_idx": [2],"temps": [300],"press": [1],"trj_freq": 10,"nsteps": 9000, "ensemble": "nvt", "_idx": "02"
},
{"sys_idx": [3],"temps": [300],"press": [1],"trj_freq": 10,"nsteps": 18000, "ensemble": "nvt", "_idx": "03"
}
],
"_comment": " 02.fp ",
"fp_style": "vasp",
"shuffle_poscar": false,
"fp_task_max": 200,
"fp_task_min": 8,
"fp_pp_path": "",
"fp_pp_files":["", "", ""],
"fp_incar": "/INCAR_methane"
}

**Comment by HuangJiameng** [comment, marked as answer]
How did you compare `energy.raw` with `energy.npy`? Do you mean that you manually did self-consistent field calculation with the input files in `02.fp` using VASP but found a great difference with results generated by DP-GEN?

**Reply by nzynpu** [reply]
Hello, sorry, maybe my expression is not clear, I have a script that can extract the .pb file and coord.raw file of the trained model to extract the total energy of the system after training and then compare with 02.fp/data.000/set.000/energy.npy.Then I found that there is a big difference , but this situation does not appear when comparing the force.

**Reply by HuangJiameng** [reply]
This problem has been discussed in WeChat before, and I would like to provide you with some suggestions for other users.
1. Is your test data in the training set? If not, this may be the case.
2. If the test data exists in the training set, please check whether the units of force and energy are written correctly. 
3. Some software will turn the orientation of coordinates. Please check it.
4. You can also scan the potential energy surface with the trained model to see the difference between your machine learning potential and quantification.
